src/main/java/io/cdap/plugin/gcp/firestore/source/FirestoreSourceConfig.java

```diff
@@ -270,8 +259,12 @@ void validateFirestoreConnection(FailureCollector collector) {
         .withConfigProperty(FirestoreConstants.PROPERTY_COLLECTION)
         .withStacktrace(e.getStackTrace());
     } catch (Exception e) {
+      collector.addFailure(e.getMessage(), "Error while connecting to Firestore.")
```

Comment:
The second one is the action that should be taken, In this case we can prefix error with `Error while connecting to Firestore: {}, e.getMessage()` and keep action null

Reply:
Done.